src/lib.rs

```diff
@@ -565,8 +614,70 @@ pub fn get_lines_from_osm(pbf: &mut OsmPbfReader) -> Vec<Line> {
         .collect()
 }
 
+pub fn get_routes_from_stop(routes: &Vec<Route>, stop_point: &StopPoint) -> Vec<Route> {
+    routes
+        .iter()
+        .filter(|route| route.contains_stop_point_id(&stop_point.id))
+        .map(|obj| obj.clone())
+        .collect()
+}
+
+pub fn categorize_stop_point(stop_point: &StopPoint, routes: Vec<Route>) -> StopPoint {
+    let mut result_sp = stop_point.clone();
+    if result_sp
+        .all_osm_tags
+        .contains("public_transport", "platform")
+    {
+        result_sp.stop_point_type = StopPointType::Platform;
+    } else if result_sp
+        .all_osm_tags
+        .contains("public_transport", "stop_position")
+    {
+        result_sp.stop_point_type = StopPointType::StopPosition;
+    } else {
+        let routes_ptv2: Vec<Route> = routes
+            .iter()
+            .filter(|r| r.all_osm_tags.contains("public_transport:version", "2"))
+            .map(|obj| obj.clone())
+            .collect();
+        warn!(
+            "categorization of stop_point {} needs pt_v2 routes. {} ptv2 routes found",
+            stop_point.id,
+            routes_ptv2.len()
+        );
```

Comment:
Should we sort the `route_ptv2` vec to be sure we always check it in the same order ?
```suggestion
        routes_ptv2.sort_by(|a, b| b.id.cmp(&a.id));
        for route in routes_ptv2 {
```

Reply:
yes, i planned to do it, but forgot :sweat_smile: 
thx